### PR TITLE
Move all ODF-related resources into feature/odf

### PR DESCRIPTION
**Pull request by larsks**
This allows us to enable/disable ODF with a single comment.

With this commit, ODF is still disabled.

### CODE REVIEW

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml [hunk 1]
```diff
@@ -7,8 +7,8 @@ resources:
 - ../common
 - ../../bundles/acct-mgt
 - ../../bundles/metallb
-# - ../../bundles/odf-external
 - ../../bundles/xdmod-reader
+# feature/odf
```
Comment (dystewart) [inline cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml:11]: Remove comment

File: cluster-scope/overlays/nerc-ocp-prod/feature/odf/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+kind: Kustomization
+commonLabels:
+  nerc.mghpcc.org/feature: odf
+
+components:
+  - ../../../../components/nerc-secret-store
+
+resources:
+  - ../../../../bundles/odf-external
+  - externalsecrets/rook-ceph-external-cluster-details.yaml
```
Comment (naved001) [inline cluster-scope/overlays/nerc-ocp-prod/feature/odf/kustomization.yaml:11]: we should add the storageclass patch to this (it is commented out right now, so wouldn't matter).

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml (hunk 1 above)
Comment (larsks) [reply cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml:11]: That comment was intentional :)

File: cluster-scope/overlays/nerc-ocp-prod/feature/odf/kustomization.yaml (hunk 2 above)
Comment (larsks) [reply cluster-scope/overlays/nerc-ocp-prod/feature/odf/kustomization.yaml:11]: Good point, I've updated the pull request.

**Review by dystewart — APPROVED** [review]

**Review by naved001 — APPROVED** [review]

**Pull request merged**